### ISSUE TITLE
Unable to switch from ALPHA to STABLE channel with electron-updater

### ISSUE THREAD

**Issue by Zagrios** [electron-userland/electron-builder#7669]
**Electron-Builder Version** : 24.4.0
**Electron-Updater Version**: 6.1.1

**Node Version**: 19.1.0

**Electron Version**: 25.3.0
**Electron Type (current, beta, nightly)**: current

**Target**: win-nsis

Hi, I have two versions of my electron app: "1.2.0-alpha.4" and "1.2.0", and I want users currently using the ALPHA version to switch to the stable version when it's available (ex: 1.2.0-alpha.4 -> 1.2.0).

Currently, for some reason, users are unable to access the stable version.

The only options I set for autoUpdater are:
```
autoUpdater.logger = log;
autoUpdater.autoDownload = false;
```
When I call the checkForUpdates() function, I got this message in the logs (even if the stable version is available and set as "latest" on GitHub):
`Update for version 1.2.0-alpha.4 is not available (latest version: 1.2.0-alpha.4, downgrade is allowed).`

I also tried modifying some properties of the autoUpdater (allowDowngrade , channel, allowPrerelease), but nothing has changed.

Thanks for your help.

**Comment by mmaietta** [comment]
Can you please try testing this fix using [patch-package](https://www.npmjs.com/package/patch-package) on your local project?
`electron-updater+6.1.3.patch`
```
diff --git a/node_modules/electron-updater/out/providers/GitHubProvider.js b/node_modules/electron-updater/out/providers/GitHubProvider.js
index 4adfd43..b185c9b 100644
--- a/node_modules/electron-updater/out/providers/GitHubProvider.js
+++ b/node_modules/electron-updater/out/providers/GitHubProvider.js
@@ -61,7 +61,7 @@ class GitHubProvider extends BaseGitHubProvider {
                         //Get Channel from this release's tag
                         const hrefChannel = ((_c = semver.prerelease(hrefTag)) === null || _c === void 0 ? void 0 : _c[0]) || null;
                         const shouldFetchVersion = !currentChannel || ["alpha", "beta"].includes(currentChannel);
-                        const isCustomChannel = !["alpha", "beta"].includes(String(hrefChannel));
+                        const isCustomChannel = hrefChannel !== null && !["alpha", "beta"].includes(String(hrefChannel));
                         // Allow moving from alpha to beta but not down
                         const channelMismatch = currentChannel === "beta" && hrefChannel === "alpha";
                         if (shouldFetchVersion && !isCustomChannel && !channelMismatch) {
```

**Comment by mmaietta** [comment]
Should be fixed in `electron-updater@6.1.4`

**Comment by Zagrios** [comment]
Relly cool thanks 🥳  sry i was too busy to try your patch
